Source/WebCore/editing/ReplaceSelectionCommand.cpp

```diff
@@ -911,18 +912,27 @@ VisiblePosition ReplaceSelectionCommand::positionAtStartOfInsertedContent() cons
 static bool handleStyleSpansBeforeInsertion(ReplacementFragment& fragment, const Position& insertionPos)
 {
     RefPtr topNode { fragment.firstChild() };
+    if (!is<HTMLSpanElement>(topNode))
+        return false;
 
     // Handling the case where we are doing Paste as Quotation or pasting into quoted content is more complicated (see handleStyleSpans)
     // and doesn't receive the optimization.
     if (isMailPasteAsQuotationNode(topNode.get()) || enclosingNodeOfType(firstPositionInOrBeforeNode(topNode.get()), isMailBlockquote, CanCrossEditingBoundary))
         return false;
 
+    // Remove style spans to follow the styles of list item when |fragment| becomes a list item.
+    // See bug - https://bugs.webkit.org/show_bug.cgi?id=127242
+    auto& wrappingStyleSpan = downcast<HTMLSpanElement>(*topNode);
+    if (isListItem(enclosingBlock(insertionPos.anchorNode()))) {
```

Comment:
This doesn't seem right. We shouldn't be removing any span element just because we've merged the subsequent paragraph to a list.

Reply:
I think the problem is when we merge the paragraph to list-items, we add another span within span on the first line item. Which this patch is trying to remove. Let me know, if it is something we should fix to align with Blink since Gecko is doing similar to what we are doing as of right now (e.g., based on the failing test added by this patch).